core/dbt/task/retry.py

```diff
@@ -0,0 +1,140 @@
+from pathlib import Path
+
+from dbt.cli.flags import Flags
+from dbt.cli.types import Command as CliCommand
+from dbt.config import RuntimeConfig
+from dbt.config.runtime import load_project, load_profile
+from dbt.contracts.results import NodeStatus
+from dbt.contracts.state import PreviousState
+from dbt.exceptions import DbtRuntimeError
+from dbt.flags import get_flags
+from dbt.graph import GraphQueue
+from dbt.task.base import ConfiguredTask
+from dbt.task.build import BuildTask
+from dbt.task.compile import CompileTask
+from dbt.task.freshness import FreshnessTask
+from dbt.task.generate import GenerateTask
+from dbt.task.run import RunTask
+from dbt.task.run_operation import RunOperationTask
+from dbt.task.seed import SeedTask
+from dbt.task.show import ShowTask
+from dbt.task.snapshot import SnapshotTask
+from dbt.task.test import TestTask
+
+RETRYABLE_STATUSES = {NodeStatus.Error, NodeStatus.Fail, NodeStatus.Skipped, NodeStatus.RuntimeErr}
+
+TASK_DICT = {
+    "build": BuildTask,
+    "compile": CompileTask,
+    "freshness": FreshnessTask,
+    "generate": GenerateTask,
+    "run": RunTask,
+    "seed": SeedTask,
+    "show": ShowTask,
+    "snapshot": SnapshotTask,
+    "test": TestTask,
+    "run-operation": RunOperationTask,
+}
+
+CMD_DICT = {
```

Comment:
Can we add two unit test to check
1. the TASK_DICT's key always matches the CMD_DICT's key?
2. for all commands that are in CLI, we either have them defined in this CMD_DICT, or it is defined in the test as something we don't support retry on.

So that next time we add new command a test would fail to tell us to see whether we should add it to here.